src/replica/bulk_load/replica_bulk_loader.cpp

```diff
@@ -609,21 +596,23 @@ void replica_bulk_loader::download_sst_file(const std::string &remote_dir,
     METRIC_VAR_INCREMENT_BY(bulk_load_download_file_bytes, f_size);
 
     // download next file
+    file_meta next_f_meta;
     {
         zauto_read_lock l(_lock);
-        if (file_index + 1 < _metadata.files.size()) {
-            f_meta = _metadata.files[file_index + 1];
+        if (!_metadata.files.empty()) {
```

Comment:
Nowthat `_metadata` is updated here, it's needed to use "write lock", i.e. `zauto_write_lock`.